### ISSUE TITLE
the -maf parameter problem

### ISSUE THREAD

**Issue by BlueIsTheWarmestColor** [voichek/kmersGWAS#99]
Dear Voichek,

A weird thing happened to me. I used two sets of parameters, which are -mac 5 and -maf 0.05 -mac 5, to chech different parameters' results. For the -maf 0.05 -mac 5, I got no significant kmer with threshold_5per 9.778720. But for the -mac 5, I got more than 800 significant kmers with  threshold_5per 9.764040. Then I checked the significant kmers' frequencies in accessions from the filter_kmers output table, which are between 0.05019 to 0.22602. So I don't know why an additional parameter -maf 0.05 would result no significant kmers, however the significant kmers' frequencies are more than 0.05.

Looking forward to your reply.

Sincerely,
Simon.

**Comment by voichek** [comment]
Dear Simon,

Interesting, if I get it correctly:
| parameters in run | threshold | # of significant _k-mers_ |
| ----------------- | ---------- | ------------------------ |
| -maf 0.05 -mac 5 | 9.778720 | 0 |
| -mac 5 | 9.764040 | 800+ |

Indeed sounds weird, first, you do see a different in the threshold 9.76 --> 9.77, are the k-mers that you get are in this range?

The MAF & MAC are combined into one MAC threshold, which is the highest of the two. You can see this number in the begining of the `associate_kmers.log` file, for example: `Effective minor allele count:	12`. Check that it make sense.

If this two ideas won't explain the difference, can you share with me the pipeline output so I can try to understand what is going on. I will remove everything of course after I finish inspecting it.

Best wishes,
Yoav

**Comment by BlueIsTheWarmestColor** [comment]
Dear Voichek,

Thanks for your patience, lately I been busy on other stuffs, sorry for replying late.

For the question I asked, I checked and found that I made mistakes on input data resulting no significant kmers.

But I do find other problems. First, when I used -mac 5 on two traits, one reported Effective minor allele count: 29, the other reported Effective minor allele count: 150. Second, when I used -mac 5 and -maf 0.05 -mac 5 on the same trait, it reported Effective minor allele count: 29 and Effective minor allele count: 143 respectively.

In conclusion, I don't know how the script decided the Effective minor allele count: 143 concretely. I feel confused why the same parameter would result different effective minor allele counts. 

Looking forword to your reply.

Sincerely,
Simon.

**Comment by voichek** [comment]
Hi Simon,

Good to hear, I was worried it was a bug.

Regarding _maf_ and _mac_, both of them have default values (_maf_ = 0.05, _mac_ = 5), which I think is the issue, so if you don't state a value it takes the default one.  Regarding your results I guess for all of them only the _maf_ has an effect anyway. The difference in number should be due to the number of accession having a phenotype values in your phenotype file and are also part of your accessions in  the _k-mers_ table. Therefore, my guess is that either you have less accessions in the phenotype file, or for some reason it can not find them in the `kmer-table.names` list, maybe due to different in naming.

In the `EMMA_perm.log` file you have a line with `EMMA_n_accessions = ` which contain the number of accession it works on, check that it makes sense. And in the `pheno.phenotypes` file it has all the accession after intersecting with the names in your _k-mers_ table.

Let me know if this solves the issue,
Best,
Yoav

**Comment by BlueIsTheWarmestColor** [comment]
Dear Voichek,

Thanks you so much for the answer. My problem has been well solved.

Sincerely,
Simon.
